### PR TITLE
feat(css/lints): Allow regex in "ignore list"

### PR DESCRIPTION
**Pull request by g-plane**
**Description:**

In Stylelint, some rules, for example [`at-rule-no-unknown`](https://stylelint.io/user-guide/rules/list/at-rule-no-unknown), allow regular expressions in configuration, and this PR implemented that feature.

**BREAKING CHANGE:**

API of `swc_css_lints` has breaking changes, because its function signature has changed.

### CODE REVIEW

**Review by alexander-akait — APPROVED** [review]
Looks good, thanks

**Review by kdy1 — CHANGES_REQUESTED** [review]
Can you change to expose `anyhow::Error` instead of `regex::Error`?

**Comment by g-plane** [comment]
Would it be better that using `thiserror`?

**Comment by kdy1** [comment]
Yes, because all errors belong to a configuration error

File: crates/swc_css_lints/src/pattern.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+use regex::Regex;
+
+#[derive(Debug)]
+pub(crate) enum NamePattern {
+    Str(String),
+    Regex(Regex),
```
Comment (kdy1) [inline crates/swc_css_lints/src/pattern.rs:6]: Can you use [CachedRegex](https://docs.rs/swc_cached/latest/swc_cached/regex/struct.CachedRegex.html) instead?

Comment (kdy1) [reply crates/swc_css_lints/src/pattern.rs:6]: I made `CachedRegex` because the configuration is typically shared among files in a project.

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_css_lints --breaking

**Review by kdy1 — APPROVED** [review]

**Pull request merged**